x-pack/plugins/alerts/common/alert.ts

```diff
@@ -68,6 +69,7 @@ export interface Alert {
   apiKey: string | null;
   apiKeyOwner: string | null;
   throttle: string | null;
+  notifyWhen: AlertNotifyWhenType | null;
```

Comment:
Changed the field from a boolean to a string type with 3 options. This aligns better. with the dropdown selector with 3 options. Allowing this to be `null` to provide backwards compatibility.

Reply:
@pmuellr I was thinking more in the context of stack monitoring, SIEM and whoever else using our APIs today. Adding a required field would break their usage (of alerts client / HTTP API) and I was thinking we could apply the same logic as the migrations to determine what value to set for these calls. Later on, 8.0, make this a mandatory field (follow up).